engine/src/io/data_provider/DummyProvider.h

```diff
@@ -34,27 +34,27 @@ class dummy_data_provider : public data_provider {
 		// does nothing
 	}
 
-	data_handle get_next() {
+	data_handle get_next(bool open_file = true) {
```

Comment:
I am really starting to dislike this logic where you tell it open th efile to set some state to true or not true. In reality what wewant here is to reset the counter after we have run get_next once of a few times. The proper way to do this is to make an api that is .reset_provider or somethign like this. It seems to be poor form to have get_next operate these two ways. Another way would be to do something like peek() instead of get_next if you wanted to get some of the data but not change the internal state. This just seems a bit confusing to me as a user.

if I saw

handle.get_next(true), handle.get_next(false) I woul dhave to go into the function signature to see what it was trying to do. If instead you did

```
handle.get_next() -- get me the next one and change state
handle.peek(3)  -- get me the next 3 and change no state
```

this becomes much clearer